src/kixi/stats/distribution.cljc

```diff
@@ -506,6 +506,59 @@
         :cljs (ISeqable
                (-seq [this] (sampleable->seq this)))))
 
+(deftype ^:no-doc Cauchy
+  [location scale]
+  p/PRandomVariable
+  (sample-1 [this rng]
+    (+ location (* scale (tan (* PI (- (rand-double rng) 0.5))))))
+  (sample-n [this n rng]
+    (default-sample-n this n rng))
+  p/PQuantile
+  (cdf [this x]
+    (+ 0.5 (/ (atan (/ (- x location) scale)) PI)))
+  (quantile [this p]
+    (+ location (* scale (tan (* PI (- p 0.5))))))
+  #?@(:clj (clojure.lang.Seqable
+            (seq [this] (sampleable->seq this)))
+      :cljs (ISeqable
+             (-seq [this] (sampleable->seq this)))))
+
+(deftype ^:no-doc Lognormal
+  [logmu logsd]
+  p/PRandomVariable
+  (sample-1 [this rng]
+    (exp (+ (* (rand-normal rng) logsd) logmu)))
+  (sample-n [this n rng]
+    (default-sample-n this n rng))
+  p/PQuantile
+  (cdf [this x]
+    (* 0.5 (+ 1 (erf (/ (- (log x) logmu)
+                           (sqrt (* 2 logsd logsd)))))))
+  (quantile [this p]
+    (exp (+ (* -1.41421356237309505 logsd (erfcinv (* 2 p))) logmu)))
```

Comment:
Checks out 👍

Reply:
I was wondering whether having a set of tests that checked for

> (cdf distrib (quantile distrib x)) = x

would be worthwhile.

Reply:
Definitely, I really like that idea